### ISSUE TITLE
Openssl Memory leak

### ISSUE THREAD

**Issue by abdessamad-zevenet** [cherokee/webserver#1217]
Hi,
Did anyone realize a growing memory usage in cherokee over time whit tls enabled and requesting a simple cgi script?, after running a memory heap tracer it seems that it's comes under ssl_do_handshake when using openssl <= 1.1. with openssl 1.1.1 the things seems to get better and at some point memory stops growing.

![screenshot from 2018-11-29 17-44-19](https://user-images.githubusercontent.com/33321464/49237639-f310b000-f3fe-11e8-9080-95ced295c9ba.png)

Also, how to get openssl statically linked ?, already tried --enable static and also adding LDFLAGS="-lssl -lcrypto -lz -ldl -static-libgcc" and no luck.

Thanks in advance.

**Comment by skinkie** [comment]
Could you compile cherokee *without* trace and see if you can reproduce it?

**Comment by abdessamad-zevenet** [comment]
Thank you for answering. 
I compile cherokee without the options "--enable-trace --enable-backtraces", I just use --with-mysql=no --with-ffmpeg=no --with-ldap=no, but also tried with default configure and same happens. What I think is that SSL session allocation are not freed correctly. Also I realized that CLEAR_LIBSSL_ERRORS leak some bytes when are no error queue when using ERR_get_error().
Only improvement I saw is when using latest libssl version.

Thanks for your help!
